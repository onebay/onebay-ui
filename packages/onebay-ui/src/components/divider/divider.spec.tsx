import { mount } from '@vue/test-utils'
import Divider from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Divider, {
    components: {},
    slots,
    props: { ...values },
  })
}

describe('Divider Snap', () => {
  it('render initial Divider', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Divider -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Divider -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Divider -- props content', () => {
    const wrapper = factory({ content: 'content' }, { default: void 0 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Divider -- props height', () => {
    const wrapper = factory({ height: '120' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Divider -- props fontColor', () => {
    const wrapper = factory({ fontColor: '#fff' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Divider -- props fontSize', () => {
    const wrapper = factory({ fontSize: '56' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Divider -- props lineColor', () => {
    const wrapper = factory({ lineColor: '#fff' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Divider -- props childen', () => {
    const wrapper = factory({ childen: 'content' }, { default: ['test'] })
    expect(wrapper.element).toMatchSnapshot()
  })
})
