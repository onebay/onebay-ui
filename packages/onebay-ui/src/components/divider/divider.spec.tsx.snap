// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Snap render Divider -- props childen 1`] = `
<div
  childen="content"
  class="ob-divider"
>
  <div
    class="ob-divider__content"
  >
    
    test
    
  </div>
  <div
    class="ob-divider__line"
  />
</div>
`;

exports[`Divider Snap render Divider -- props className 1`] = `
<div
  class="ob-divider test"
>
  <div
    class="ob-divider__content"
  >
    
    
  </div>
  <div
    class="ob-divider__line"
  />
</div>
`;

exports[`Divider Snap render Divider -- props content 1`] = `
<div
  class="ob-divider"
>
  <div
    class="ob-divider__content"
  >
    content
  </div>
  <div
    class="ob-divider__line"
  />
</div>
`;

exports[`Divider Snap render Divider -- props customStyle 1`] = `
<div
  class="ob-divider"
  style=""
>
  <div
    class="ob-divider__content"
  >
    
    
  </div>
  <div
    class="ob-divider__line"
  />
</div>
`;

exports[`Divider Snap render Divider -- props fontColor 1`] = `
<div
  class="ob-divider"
>
  <div
    class="ob-divider__content"
    style="color: rgb(255, 255, 255);"
  >
    
    
  </div>
  <div
    class="ob-divider__line"
  />
</div>
`;

exports[`Divider Snap render Divider -- props fontSize 1`] = `
<div
  class="ob-divider"
>
  <div
    class="ob-divider__content"
    style="font-size: 56;"
  >
    
    
  </div>
  <div
    class="ob-divider__line"
  />
</div>
`;

exports[`Divider Snap render Divider -- props height 1`] = `
<div
  class="ob-divider"
  style=""
>
  <div
    class="ob-divider__content"
  >
    
    
  </div>
  <div
    class="ob-divider__line"
  />
</div>
`;

exports[`Divider Snap render Divider -- props lineColor 1`] = `
<div
  class="ob-divider"
>
  <div
    class="ob-divider__content"
  >
    
    
  </div>
  <div
    class="ob-divider__line"
    style="background-color: rgb(255, 255, 255);"
  />
</div>
`;

exports[`Divider Snap render initial Divider 1`] = `
<div
  class="ob-divider"
>
  <div
    class="ob-divider__content"
  >
    
    
  </div>
  <div
    class="ob-divider__line"
  />
</div>
`;
